### PR TITLE
other(element-template-generator): add static feel into element template for boolean properties

### PR DESCRIPTION
**Pull request by mathias-vandaele**
## Description

It was required to add `"feel" :  "static"` to all element templates for boolean, I did this also for `Number`
 
## Related issues

closes https://github.com/camunda/team-connectors/issues/999

## Checklist

- [ ] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java [hunk 1]
```diff
@@ -182,6 +118,72 @@ private static void validateCondition(
...
+    if (annotation == null) {
+      return;
+    }
+    builder.optional(FieldProcessor.isOptional(field));
+
+    if (!(builder instanceof DropdownPropertyBuilder)) {
+      if (Boolean.class.equals(field.getType()) || boolean.class.equals(field.getType())) {
+        builder.feel(Property.FeelMode.staticFeel);
+      } else if (annotation.feel() == Property.FeelMode.system_default) {
+        builder.feel(determineDefaultFeelModeBasedOnContext(context));
```
Comment (sbuettner) [inline element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java:134]: Shouldnt this all happen in `determineDefaultFeelModeBasedOnContext`?

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java [hunk 2]
```diff
@@ -182,6 +118,72 @@ private static void validateCondition(
...
+        case Boolean:
+          builder.value(Boolean.parseBoolean(value));
+          break;
+        case String:
+          builder.value(value);
+          break;
+        default:
+          throw new IllegalStateException("Unexpected value: " + annotation.defaultValueType());
+      }
+    }
```
[the remark this answers is not in the document]
Comment (sbuettner) [reply element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java:158]: @mathias-vandaele Also curious: Cant we deduct the default value type based on the `PropertyType`? Or am I missing something?

**Comment by sbuettner** [comment]
@mathias-vandaele Can you double check whether we also can support numbers now?

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java (hunk 1 above)
Comment (mathias-vandaele) [reply element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java:134]: Doing it

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mathias-vandaele) [reply element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java:158]: We could yes the `PropertyType`, but this is something the user could forget to set, isn't better to use the real class value of the field ?

**Comment by mathias-vandaele** [comment]
> @mathias-vandaele Can you double check whether we also can support numbers now?

Could you elaborate on this @sbuettner ?

**Comment by sbuettner** [comment]
@mathias-vandaele The issue mentions that the static feel option is also used for number support and I was wondering whether we now support numbers as well as types.

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbuettner) [reply element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java:158]: I thought we need to determine `PropertyType` based on the actual type. 

In general: Would also be interesting to know we need the `DefaultValueType` at all. cc @chillleader

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/TemplatePropertiesUtil.java [hunk 3]
```diff
@@ -276,16 +267,32 @@ private static List<DropdownModel> createDropdownModelList(Class<?> enumType) {
...
+    }
+    return Number.class.isAssignableFrom(clazz);
+  }
+
   private static PropertyBuilder createPropertyBuilder(Field field, TemplateProperty annotation) {
     PropertyType type;
     List<DropdownModel> dropdownChoices = new ArrayList<>();
 
-    if (field.getType() == Boolean.class) {
+    if (field.getType() == Boolean.class || field.getType() == boolean.class) {
```
Comment (johnBgood) [inline element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/TemplatePropertiesUtil.java:288]:
```suggestion
    if (field.getType() == Boolean.class || field.getType() == Boolean.TYPE) {
```

**Review by johnBgood — DISMISSED** [review]
🥇

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*

**Review by johnBgood — APPROVED** [review]
Great! I don't know if we have tests in the ETG, but if so we might want to add some about this new feature.

**Review by Szik — APPROVED** [review]
